### DISCUSSION TITLE
'brew config' displays CLT: N/A

### DISCUSSION THREAD

**Discussion by robertleeblairjr** [Homebrew/discussions#15] [Everyday usage] [answered]
My computer is a 16" MacBook Pro on macOS 10.15.7 and Xcode 12.0.1. Prior to the update for the latest macOS and Xcode, `brew config` displayed the correct Command-Line Tools version. I have the CLT installed via Xcode, which was installed via the App Store. `xcode-select --print-path' shows "/Applications/Xcode.app/Contents/Developer" as the correct path and 'xcode-select --version' shows "version 2373". Opening Xcode > Preferences and navigating to the tab Locations shows that the Command-Line Tools are installed with the correct path. I'm fairly certain this is a bug which I'll report an issue with but wanted to confirm if anyone else is having this same problem. I've seen this occur in the past and it either resolved with 'brew doctor', 'brew update-reset' or from some other update to Homebrew. Any one have suggestions other than reinstall? Below is the output from 'brew config'.

> HOMEBREW_VERSION: 2.5.2-41-g9fbe13c
ORIGIN: https://github.com/Homebrew/brew
HEAD: 9fbe13c17d3a781f62992725eeb22b24edc69a2f
Last commit: 2 days ago
Core tap ORIGIN: https://github.com/Homebrew/homebrew-core
Core tap HEAD: 3be15ade1a7f0dd37c757d9992c791d0bb7d5264
Core tap last commit: 16 minutes ago
Core tap branch: master
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.qsyuB7GKIH/org.macosforge.xquartz:0
HOMEBREW_EDITOR: /usr/local/bin/nano
HOMEBREW_GITHUB_API_TOKEN: set
HOMEBREW_MAKE_JOBS: 16
Homebrew Ruby: 2.6.3 => /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/bin/ruby
CPU: 16-core 64-bit kabylake
Clang: 12.0 build 1200
Git: 2.28.0 => /usr/local/bin/git
Curl: 7.64.1 => /usr/bin/curl
Java: 15, 1.8.0_265
macOS: 10.15.7-x86_64
CLT: N/A
Xcode: 12.0.1
XQuartz: 2.7.11 => /opt/X11

**Comment by SMillerDev** [comment, marked as answer]
As far as I know the XCode CLT is not the same as the normal CLT so that would make the current display a truthful representation of your system

**Reply by robertleeblairjr** [reply]
Ah! I assumed that whenever opening Xcode for the first time and using the xcode-select from terminal installs the Command-Line Tools. Attempting to install CLT after they are installed using 'sudo xcode-select --install" replies with that they are already installed. Installing any formulae, casks, updates, style/audit checks, tests, etc. all work without a problem. It's just the output from 'brew config' which shows the "N/A". Can you explain to me why all of these work but the config file doesn't show a version number?

**Reply by SMillerDev** [reply]
Because it's just a representation of what's installed. You don't have the macOS package that is "Command Line Tools" so it doesn't have a version either. That doesn't mean that you can't use brew or that things are broken. Just that it's not there.

**Reply by robertleeblairjr** [reply]
For an IDE to be almost 12GB, you would think it included all of the tools which are packaged separately. It's even listed for their developer documentation that downloading and installing Xcode will come with the embedded command line tools. Also, the command line tools package description states that if you have Xcode installed, then this package is included. Thanks for your help though.
<img width="1436" alt="Screen Shot Apple Developer Downloads" src="https://user-images.githubusercontent.com/12276457/94483345-56f19480-01a0-11eb-9817-848429ba4b0c.png">
